tests/all/integration/api/attendee/test_attendee_api.py

```diff
@@ -129,6 +143,151 @@ def test_edit_attendee_required_fields(db, client, jwt):
     assert attendee.tax_business_info == 'Hello'
 
 
+def get_complex_custom_form_attendee(db):
+    attendee = get_minimal_attendee(db)
+    CustomForms(
+        event=attendee.event,
+        form='attendee',
+        field_identifier='jobTitle',
+        type='text',
+        is_included=True,
+        is_required=True,
+    )
+    CustomForms(
+        event=attendee.event,
+        form='attendee',
+        field_identifier='bestFriend',
+        name='Best Friend',
+        type='text',
+        is_included=True,
+        is_required=True,
+        is_complex=True,
+    )
+    db.session.commit()
+
+    return attendee
+
+
+def test_custom_form_complex_fields_missing_required(db, client, jwt):
+    attendee = get_complex_custom_form_attendee(db)
+
+    data = json.dumps(
+        {
+            'data': {
+                'type': 'attendee',
+                'id': str(attendee.id),
+                "attributes": {"firstname": "Areeb", "lastname": "Jamal"},
+            }
+        }
+    )
+
+    response = client.patch(
+        f'/v1/attendees/{attendee.id}',
+        content_type='application/vnd.api+json',
+        headers=jwt,
+        data=data,
+    )
+
+    db.session.refresh(attendee)
+
+    assert response.status_code == 422
+    assert json.loads(response.data) == {
+        'errors': [
+            {
+                'detail': "Missing required fields ['best_friend', 'job_title']",
+                'source': {'pointer': '/data/attributes'},
+                'status': 422,
+                'title': 'Unprocessable Entity',
+            }
+        ],
+        'jsonapi': {'version': '1.0'},
+    }
+
+    assert attendee.firstname != 'Areeb'
+    assert attendee.lastname != 'Jamal'
+    assert attendee.complex_field_values is None
+
+
+def test_custom_form_complex_fields_missing_required_one(db, client, jwt):
```

Comment:
Codacy found an issue: [Function name "test_custom_form_complex_fields_missing_required_one" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5740141)